ipatests/test_webui/ui_driver.py

```diff
@@ -1700,32 +1749,53 @@ def get_t_vals(t):
             # add multiple at once and test table delete button
             self.add_table_associations(table, keys, delete=True)
 
-    def add_sshkey_to_user(self, user, ssh_key):
+    def add_sshkey_to_user(self, ssh_keys, user='admin', navigate=True,
+                           save=True):
         """
         Add ssh public key to particular user
 
+        ssh_keys (list): public ssh key(s)
         user (str): user to add the key to
-        ssh_key (str): public ssh key
         """
-        self.navigate_to_entity('user')
-        self.navigate_to_record(user)
 
-        ssh_pub = 'div[name="ipasshpubkey"] button[name="add"]'
-        self.find(ssh_pub, By.CSS_SELECTOR).click()
-        self.wait()
-        self.driver.switch_to.active_element.send_keys(ssh_key)
-        self.dialog_button_click('update')
-        self.facet_button_click('save')
+        if type(ssh_keys) is not list:
+            ssh_keys = [ssh_keys]
+
+        if navigate:
+            self.navigate_to_entity('user')
```

Comment:
I know it was in the original method. But it would be better if instead of string a constant would be used. I'm also not sure about navigation in this method, but it was here before this patch so could be kept here.

Reply:
In https://github.com/freeipa/freeipa/pull/1679 you suggested to make navigation configurable however I really do not mind to change it.

Reply:
So let's keep it here and make the method more general to work for host and id views as we talked offline.